conda_forge_tick/migrators/extra_jinj2a_keys_cleanup.py

```diff
@@ -0,0 +1,67 @@
+import os
+import typing
+from typing import Any
+from ruamel.yaml import YAML
+
+from conda_forge_tick.xonsh_utils import indir
+from conda_forge_tick.migrators.core import MiniMigrator
+from conda_forge_tick.recipe_parser import CondaMetaYAML
+
+if typing.TYPE_CHECKING:
+    from ..migrators_types import AttrsTypedDict
+
+
+class ExtraJinja2KeysCleanup(MiniMigrator):
+    """Remove extra Jinja2 keys that make things hard for the version migrator.
+
+    Variables removed:
+
+    - ``hash_type``: This is sometimes used as a key in the ``source`` section
+      to map the hash_type (ex. sha256) to the hash.
+    - ``hash_value``: This is unnecessary since it is typically only used once
+      in a YAML recipe.
+    - ``file_ext``: This is unnecessary and just makes things harder to read.
+      It may also be causing trouble for the version migrator.
+
+    """
+
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        """Check the raw YAML to find jinja variables to deal with."""
+        raw_yaml = attrs['raw_meta_yaml']
+        if '{% set file_ext' in raw_yaml or '{% set hash_type' in raw_yaml:
+            return False
+        return True
+
+    def _replace_jinja_key(self, key_name, lines):
+        """Replace any usage of and the definition of a jinja2 variable."""
+        key_val = None
+        var_use = "{{ " + key_name + " }}"
+        var_def = "{% set " + key_name
+        for line in lines:
+            if line.startswith(var_def):
+                quote1_idx = line.find('"')
+                quote2_idx = line.rfind('"')
+                key_val = line[quote1_idx+1:quote2_idx]
+                continue
+            if key_val is not None:
+                # in a string literal
+                line = line.replace("'" + var_use + "'", key_val)
```

Comment:
I am nervous about the extra quotes here given that a string could be enclosed in double quotes. jinja2 only does string replace so I would remove the quotes.